machine_learning/data_downloader.py

```
@@ -0,0 +1,160 @@
+import argparse
+import requests
+from urllib.parse import urlparse
+from web_monitoring.db import Client
+import pandas as pd
+import os
+import time
+from docopt import docopt
+import warnings
+
+def parse_url(url):
+    """
+    Parse Versionista change urls to get version ids
+    https://versionista.com/{site id}/{page id}/{to version id}:{from version id}
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    try:
+        result = urlparse(url)
+        ids = [element for element in result.path.split('/') if element]
+        site_id = ids[0]
+        page_id = ids[1]
+        version_ids = ids[2].split(':')
+        to_version_id = version_ids[0]
+        from_version_id = version_ids[1]
+        ids = {'site_id': site_id, 'page_id': page_id,
+               'from_version_id': from_version_id, 'to_version_id': to_version_id}
+        return ids
+    except:
+        warnings.warn('Url "' + url + '" is not in the required format and no data will be downloaded for this row', Warning)
```

Why not let this raise so that it's easier to interactively debug?